backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -121,21 +119,24 @@ void cuda_memcpy_async_gpu_to_gpu(void *dest, void *src, uint64_t size,
     return;
   cudaPointerAttributes attr_dest;
   check_cuda_error(cudaPointerGetAttributes(&attr_dest, dest));
-  if (attr_dest.device != gpu_index && attr_dest.type != cudaMemoryTypeDevice) {
+  if (attr_dest.type != cudaMemoryTypeDevice) {
     PANIC("Cuda error: invalid dest device pointer in copy from GPU to GPU.")
   }
   cudaPointerAttributes attr_src;
   check_cuda_error(cudaPointerGetAttributes(&attr_src, src));
-  if (attr_src.device != gpu_index && attr_src.type != cudaMemoryTypeDevice) {
+  if (attr_src.type != cudaMemoryTypeDevice) {
     PANIC("Cuda error: invalid src device pointer in copy from GPU to GPU.")
   }
-  if (attr_src.device != attr_dest.device) {
-    PANIC("Cuda error: different devices specified in copy from GPU to GPU.")
-  }
-
   check_cuda_error(cudaSetDevice(gpu_index));
+  // if (attr_src.device == attr_dest.device) {
   check_cuda_error(
       cudaMemcpyAsync(dest, src, size, cudaMemcpyDeviceToDevice, stream));
+  //} else {
+  //  printf("src device: %d, dest device: %d\n", attr_src.device,
+  //  attr_dest.device); check_cuda_error(cudaMemcpyPeerAsync(dest,
+  //  attr_dest.device, src,
+  //                                       attr_src.device, size, stream));
+  //}
```

Comment:
Maybe I could uncomment those :thinking: or I just remove it altogether.

Reply:
I like the idea here. I always feel non-intuitive that we need two different methods to copy data within a device or between different devices. I would uncomment these.